File: test/data/usecase/src/py/VernalizationProgress.py
if (IsVernalizable==1 and Vernaprog < 1):
    TT = DeltaTT
        
    if (TT >= MinTvern and TT <= IntTvern):
        Vernaprog = Vernaprog + VAI * TT + VBEE
          
    if (TT > IntTvern):
        maxVernaProg = VAI * IntTvern + VBEE
        DLverna = max(MinDL, min(MaxDL, DayLength))
        Vernaprog += max(0, maxVernaProg * (1 + ((IntTvern - TT) / (MaxTvern - IntTvern)) * ((DLverna - MinDL) / (MaxDL - MinDL))))
            
    primordno = 2.0 * LeafNumber + PNini
        
    minLeafNumber = MinFinalNumber
    if (Vernaprog >= 1.0 or primordno >= AMXLFNO):
            
            
        MinFinalNumber = max(primordno, MinFinalNumber)
        calendarMoments.append("EndVernalisation")
        calendarCumuls.append(cumulTT) 
        calendarDates.append(currentdate)        
        Vernaprog = max(1, Vernaprog)
            
            
    else:
            
        potlfno = AMXLFNO - (AMXLFNO - minLeafNumber) * Vernaprog
        if (primordno >= potlfno):
                
                
            MinFinalNumber = max((potlfno + primordno) / 2.0, MinFinalNumber)
            Vernaprog = max(1, Vernaprog)
            calendarMoments.append("EndVernalisation")
            calendarCumuls.append(cumulTT) 
            calendarDates.append(currentdate)

             

File: test/data/usecase/src/cs/VernalizationProgress.cs

if (IsVernalizable==1 && Vernaprog1 < 1)
{
	double TT = DeltaTT; // other sirius versions use previous temperature value

	if (TT >= MinTvern && TT <= IntTvern)
	{
		Vernaprog = Vernaprog + VAI * TT + VBEE;
	}
	else
	{
		Vernaprog = Vernaprog;
	}
	if (TT > IntTvern)
	{
		double maxVernaProg = VAI * IntTvern + VBEE;
		double DLverna = Math.Max(MinDL, Math.Min(MaxDL, DayLength));
		Vernaprog += Math.Max(0, maxVernaProg * (1 + ((IntTvern - TT) / (MaxTvern - IntTvern)) * ((DLverna - MinDL) / (MaxDL - MinDL))));
	}
			
	double primordno = 2.0 * LeafNumber + PNini;
	double minLeafNumber = MinFinalNumber;
	if (Vernaprog >= 1.0 || primordno >= AMXLFNO)
	{
		MinFinalNumber = Math.Max(primordno, MinFinalNumber);
		calendarMoments.add("EndVernalisation");
        calendarCumuls.add(cumulTT) ;
        calendarDates.add(currentdate);  

	}
	else
	{
		double potlfno = AMXLFNO - (AMXLFNO - minLeafNumber) * Vernaprog;
		if (primordno >= potlfno)
		{
			MinFinalNumber = Math.Max((potlfno + primordno) / 2.0, MinFinalNumber);	
			calendarMoments.add("EndVernalisation");
        	calendarCumuls.add(cumulTT);
        	calendarDates.add(currentdate);  					
			Vernaprog = Math.Max(1, Vernaprog);
		}

	}
}
